### ISSUE TITLE
Enhancement request - Make new SID generation optional during Windows OS customization

### ISSUE THREAD

**Issue by jason-azze** [terraform-providers/terraform-provider-vsphere#346]
### vSphere Provider Version
provider.vsphere v1.1.1

### Affected Resource(s)
```
resource "vsphere_virtual_machine" "vm" {
  ...

  clone {
    ...

    customize {
      ...

      windows_options {
```
The vSphere client's New VM Guest Customization Spec wizard makes generating a new Security ID (SID) for Windows guests optional. (And the default setting is to NOT generate a new SID.)
![generate-new-sid](https://user-images.githubusercontent.com/1788669/34832224-432952f4-f6b7-11e7-8d49-d87f482b76ba.png)

It would be nice to have this option available in Terraform.

There are times when you want truly identical clones made from a template, including maintaining identical SIDs for local user accounts. (In my case these local users own data on a drive that will be attached later during a CI build pipeline process.) But a new machine SID means the local users also get new SIDs.

(In fact, the only reason I'm using OS customization at all is so I can get VMs provisioned with unique names and have the guest OS renamed to match the VM name.)

**Comment by jason-azze** [comment]
@vancluever It looks like this is a blocker for me. I'm stuck with conflicting requirements. I need the guest OS naming magic provided by the VM customization process, but the accompanying mandatory SID change for Windows ruins local user account permissions that we rely on. 

Is there an obvious strategy for getting the guest OS hostname to match the VM name that I'm missing, which doesn't involve using VM customization?

**Comment by vancluever** [comment]
Hey @jason-azze,

I don't think it's possible using any current method - I was thinking of maybe if you could do it using `windows_sysprep_text`, but only if the generalization options are exposed through the sysprep.inf file, and I'm not too sure if they are.

For now, you might have to do your customization through a provisioner.

We will be exposing this though, so I will leave this open until we do that!

**Comment by jason-azze** [comment]
Thanks, @vancluever. Your hint reminded me of some work I did with `remote-exec` back before I tried making sysprep work.

For posterity, the following code worked for renaming a Windows 7 guest OS that has Cygwin SSHD installed. (I did not try the winrm connection method for this provisioner, but that might be a better bet if you have it turned on.)

```
  provisioner "remote-exec" {
    inline = [
      "cmd /C wmic computersystem where caption=\\'w7-template\\' rename \\'${vsphere_virtual_machine.windows.name}\\' > /home/test/remote.log 2>&1",
      "shutdown /r /t 30",
    ]

    connection {
      type     = "ssh"
      user     = "administrator"
      password = "${var.admin_password_value}"
    }
  }
```
The user does not have to be the actual Administrator, but might need to be a member of Local Administrators.

Note the double escapes around the single quotes and the output redirect on the wmic command. I need to open an issue against TF core for that. TF can't parse the output of the successful wmic command, and so just keeps up with "Still creating..." forever until you interrupt it.
